spec/support/authenticated_session_helper.rb

```diff
@@ -9,7 +9,7 @@ def sign_in(user = FactoryBot.build(:user, :loa3), token = nil, raw = false)
     if raw
       Rails::SessionCookie::App.new(session_object.to_hash, session_options).session_cookie
     elsif cookies.is_a?(ActionDispatch::Cookies::CookieJar)
-      request.session = session_object.to_hash
+      request.session = ActionController::TestSession.new(session_object.to_hash)
```

Comment:
This took quite awhile to come up with this simple solution. Certain areas of the app require a `Session` object, so unless we wildly patched the `TestSession` class, it wouldn't cleanly work. By only using a TestSession object here (using the attrs of the original session), we were able to satisfy all tests

Reply:
This seems reasonable 👍